src/main/java/org/opensearch/security/action/tenancy/MultiTenancyConfigurationsUpdateTransportAction.java

```diff
@@ -0,0 +1,146 @@
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+public class MultiTenancyConfigurationsUpdateTransportAction extends HandledTransportAction<MultiTenancyConfigurationsUpdateRequest, MultiTenancyConfigurationsRetrieveResponse> {
+
+    private static final Logger log = LogManager.getLogger(org.opensearch.security.action.tenancy.MultiTenancyConfigurationsUpdateTransportAction.class);
+
+    private final String securityIndex;
+    private final ConfigurationRepository config;
+    private final Client client;
+    private final ThreadPool pool;
+
+    @Inject
+    public MultiTenancyConfigurationsUpdateTransportAction(final Settings settings,
+                                             final TransportService transportService,
+                                             final ActionFilters actionFilters,
+                                             final ConfigurationRepository config,
+                                             final ThreadPool pool,
+                                             final Client client) {
+        super(MultiTenancyConfigurationsUpdateAction.NAME, transportService, actionFilters, MultiTenancyConfigurationsUpdateRequest::new);
+
+        this.securityIndex = settings.get(ConfigConstants.SECURITY_CONFIG_INDEX_NAME, ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+
+        this.config = config;
+        this.client = client;
+        this.pool = pool;
+    }
+
+    /** Load the configuration from the security index and return a copy */
+    protected final SecurityDynamicConfiguration<?> load() {
+        return config.getConfigurationsFromIndex(Collections.singleton(CType.CONFIG), false).get(CType.CONFIG).deepClone();
+    }
+
+    public Set<String> getAllConfiguredTenantNames() {
+
+        return this.config.getConfiguration(CType.TENANTS).getCEntries().keySet();
+    }
+
+    protected void validate(ConfigV7 updatedConfig) throws RuntimeException {
+        if(!updatedConfig.dynamic.kibana.private_tenant_enabled && (updatedConfig.dynamic.kibana.default_tenant).equals(ConfigConstants.MULTITENANCY_PRIVATE_TENANT_NAME) ) {
+            throw new IllegalArgumentException("Private tenant can not be disabled if it is the default tenant.");
+        }
+
+        if( (updatedConfig.dynamic.kibana.default_tenant).equals(ConfigConstants.MULTITENANCY_GLOBAL_TENANT_DEFAULT_NAME) ||
+                (updatedConfig.dynamic.kibana.default_tenant).equals(ConfigConstants.MULTITENANCY_GLOBAL_TENANT_NAME) ||
+                (updatedConfig.dynamic.kibana.default_tenant).equals(ConfigConstants.MULTITENANCY_PRIVATE_TENANT_NAME) ) {
+            return;
+        }
+
+        Set<String> availableTenants = getAllConfiguredTenantNames();
+
+        if(!availableTenants.contains(updatedConfig.dynamic.kibana.default_tenant)){
+            throw new IllegalArgumentException("Default tenant should be selected from one of the available tenants.");
```

Comment:
The error message should include the name of the tenant that was not found for troubleshooting

Reply:
Cool.